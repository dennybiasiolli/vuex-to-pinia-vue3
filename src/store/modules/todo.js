import axios from 'axios'
import { GET_TODO_LIST } from './todo-types'

export const defaultState = {
  todoList: [],
  todoListLoading: false,
  todoListLoadingError: null,
}

export const mutations = {
  [`${GET_TODO_LIST}_REQUEST`](state) {
    state.todoListLoading = true
    state.todoListLoadingError = null
  },
  [`${GET_TODO_LIST}_SUCCESS`](state, todoList) {
    state.todoListLoading = false
    state.todoListLoadingError = null
    state.todoList = todoList
  },
  [`${GET_TODO_LIST}_FAILURE`](state, error) {
    state.todoListLoading = false
    state.todoListLoadingError = error
    state.todoList = []
  },
}

export const actions = {
  async [GET_TODO_LIST]({ commit }, params) {
    commit(`${GET_TODO_LIST}_REQUEST`)
    try {
      const { data } = await axios.get(`data/todoList.json`, { params })
      commit(`${GET_TODO_LIST}_SUCCESS`, data)
    } catch (err) {
      commit(`${GET_TODO_LIST}_FAILURE`, err)
    }
  },
}

export default {
  namespaced: true,
  state: () => ({ ...defaultState }),
  mutations,
  actions,
}
