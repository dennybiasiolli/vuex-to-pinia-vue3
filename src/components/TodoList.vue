<script setup>
import { computed } from 'vue'
import { useStore } from 'vuex'
import { GET_TODO_LIST } from '@/store/modules/todo-types'

const store = useStore()
const todoList = computed(() => store.state.todo.todoList)
store.dispatch(`todo/${GET_TODO_LIST}`)
</script>

<template>
  <div v-for="todoItem of todoList" :key="todoItem.id">
    <input type="checkbox" v-model="todoItem.completed" />
    <span v-text="todoItem.name"></span>
  </div>
</template>
