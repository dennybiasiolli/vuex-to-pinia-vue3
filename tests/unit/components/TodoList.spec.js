import { describe, test, beforeEach, expect, vi } from 'vitest'
import { createStore } from 'vuex'
import { shallowMount } from '@vue/test-utils';
import TodoList from '@/components/TodoList.vue'
import { GET_TODO_LIST } from '@/store/modules/todo-types'

describe('TodoList.vue', () => {
  let store;
  let todoActions = {
    [GET_TODO_LIST]: vi.fn(),
  }
  beforeEach(() => {
    store = createStore({
      modules: {
        todo: {
          namespaced: true,
          state: () => ({
            todoList: [
              { id: 1, name: 'test', completed: true },
              { id: 2, name: 'test2', completed: false },
            ],
            todoListLoading: false,
            todoListLoadingError: null,
          }),
          actions: todoActions,
        },
      },
    })
  })
  test('should respect the snapshot', () => {
    const wrapper = shallowMount(TodoList, { global: { plugins: [store] } })
    expect(wrapper.element).toMatchSnapshot()
  })
})
