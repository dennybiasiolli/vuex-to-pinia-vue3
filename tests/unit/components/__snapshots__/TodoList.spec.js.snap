// Vitest Snapshot v1

exports[`TodoList.vue > should respect the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <div>
    <input
      type="checkbox"
    />
    <span>
      test
    </span>
  </div>
  <div>
    <input
      type="checkbox"
    />
    <span>
      test2
    </span>
  </div>
  
</div>
`;
